frontend/src/app/specification-search-detail/specification-search-detail.component.ts

```diff
@@ -12,21 +12,27 @@ import {ActivatedRoute} from '@angular/router';
 export class SpecificationSearchDetailComponent implements OnInit {
   services: Service[];
   searchString: string;
+  searchComplete = false;
 
   constructor(private serviceStore: ServiceStore, private activatedRoute: ActivatedRoute) {
   }
 
   ngOnInit() {
     this.activatedRoute.params.subscribe(params => {
       this.serviceStore.searchForServices(params['searchString'])
-        .subscribe((data: Service[]) => this.services = data);
+        .subscribe((data: Service[]) => {
+          this.services = data;
+          this.searchComplete = true;
+        });
       this.searchString = params['searchString'];
     });
   }
 
   public async searchServices(event) {
-    if (event.keyCode === 13) {
-      this.serviceStore.searchForServices(this.searchString).subscribe((data: Service[]) => this.services = data);
+    if (event.keyCode === 13 && this.searchString !== undefined && this.searchString !== '') {
```

Comment:
Should we show a warning for the other cases? So that the user doesn't wonder why hitting enter does not work.